### ISSUE TITLE
CGemv occasionally returns NAN on valid inputs

### ISSUE THREAD

**Issue by adefazio** [Theano/Theano#1569]
Very occasionally, The CGemv op is returning NAN, when its inputs are completely valid. This only occurs when device=cpu, never when run on the GPU. I'm using 64 bit windows. The error is not caused by the blas implementation as I have tried two different blas libraries (OpenBlas and AMD's ACML).
I've run the code with the detect NAN mode on, which is how I know its the CGemv op. The output from a debug session is below, where I examine the inputs/outputs at a breakpoint within the detect_nan function. None of the inputs are NAN or INF, but the output is NAN. 

Unfortunately, the most mimimal test case I can come up with is not very minimal:
https://gist.github.com/adefazio/7081767
Running this, most of the time, causes the error. But it's completely random. If it doesn't occur, try running in a different console window. It probably only occurs in 64 bit windows as well.

```
(Pdb) print node
CGemv{no_inplace}(TensorConstant{(1L,) of 0.0}, TensorConstant{1.0}, InplaceDimShuffle{x,0}.0, <TensorType(float32, vector)>, TensorConstant{0.0})

(Pdb) print fn.inputs[0]
[array([ 0.], dtype=float32)]
(Pdb) print fn.inputs[1]
[array(1.0, dtype=float32)]
(Pdb) print fn.inputs[2]
[array([[  2.03404561e-05,   2.39333585e-05,   1.19035622e-05, ...,
          1.11312575e-05,   6.02677064e-05,   3.35221559e-01]], dtype=float32)]
(Pdb) print fn.inputs[3]
[array([ 0.,  0.,  0., ...,  0.,  0.,  1.], dtype=float32)]
(Pdb) print fn.inputs[4]
[array(0.0, dtype=float32)]

(Pdb) print fn.inputs[1][0].shape
()
(Pdb) print fn.inputs[2][0].shape
(1L, 12545L)
(Pdb) print fn.inputs[3][0].shape
(12545L,)
(Pdb) print fn.inputs[4][0].shape
()

(Pdb) print fn.outputs
[[array([ nan], dtype=float32)]]

(Pdb) numpy.isnan(fn.inputs[2][0]).any()
False
(Pdb) numpy.isnan(fn.inputs[3][0]).any()
False

(Pdb) numpy.isinf(fn.inputs[2][0]).any()
False
(Pdb) numpy.isinf(fn.inputs[3][0]).any()
False


```

**Comment by lamblin** [comment]
I think this bug has been fixed by #1936. Please reopen if it is not the case.

**Comment by lamblin** [comment]
I think this bug has been fixed by #1936. Please reopen if it is not the case.

**Comment by lamblin** [comment]
I think this bug has been fixed by #1936. Please reopen if it is not the case.
